### PR TITLE
refactor: create gallery NFTs

### PR DESCRIPTION
**Pull request by shahin-hq**
## Summary

https://app.clickup.com/t/86793qcn4

This PR refactors the `useGalleryNfts` hook to be responsible for fetching and searching NFTs. In short, I removed logic that fetches NFTs in backend  and passes to the frontend components because the NFT slider doesn't really use the preloaded data. Removing backend fetching is a good idea because of the reasons below:
- Makes page loads faster (let's postpone NFT fetching until we really have to)
- It will make things a little bit simpler as we won't have to sync data: pipe backend data to frontend, instead frontend will be responsible for fetching and displaying data
- As a next step we can utilize `react-query` to cache data  - we don't have to fetch NFTs every time user opens the slider, we could cache and reuse

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [x] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

File: resources/js/Components/Galleries/Hooks/useGalleryNtfs.tsx [hunk 1]
```diff
@@ -204,10 +198,6 @@ export const useGalleryNtfs = ({
     };
 
     const searchNfts = async (query?: string): Promise<void> => {
-        if (!isTruthy(pageMeta.first_page_url)) {
-            throw new Error("[searchNfts] First page url is not defined.");
```
Comment (shahin-hq) [inline resources/js/Components/Galleries/Hooks/useGalleryNtfs.tsx:208]: `first_page_url` is always there, so we can remove this check.

File: resources/js/Components/Galleries/Hooks/useGallerysNftsContext.test.tsx
Comment (shahin-hq) [inline resources/js/Components/Galleries/Hooks/useGallerysNftsContext.test.tsx:1]: Fixing broken tests, nothing new.

File: app/Http/Controllers/MyGalleryController.php [hunk 2]
```diff
@@ -40,24 +36,12 @@ public function create(Request $request): Response|RedirectResponse
...
-
         return Inertia::render('Galleries/MyGalleries/Create', [
             'title' => trans('metatags.my_galleries.create.title'),
-            'nfts' => GalleryNftData::collection($nfts),
-            'collections' => new GalleryCollectionsData(GalleryCollectionData::collection($collections)),
-            'nftsPerPage' => $nftsPerPage,
+            'nftsPerPage' => (int) config('dashbrd.gallery.pagination.nfts_per_page'),
+            'collectionsPerPage' => (int) config('dashbrd.gallery.pagination.collections_per_page'),
             'nftLimit' => config('dashbrd.gallery.nft_limit'),
+            'nftCount' => $user->nfts->count(),
```
Comment (crnkovic) [inline app/Http/Controllers/MyGalleryController.php:44]: is there a reason you don't use query builder to get the count? `$user->nfts()->count()`. your way it loads all NFTs that user owns in a memory

File: app/Http/Controllers/MyGalleryController.php [hunk 3]
```diff
@@ -80,28 +64,16 @@ public function edit(Request $request, Gallery $gallery): Response
...
-            'collections' => new GalleryCollectionsData(GalleryCollectionData::collection($collections)),
             'gallery' => GalleryData::fromModel(
                 gallery: $gallery,
                 limit: config('dashbrd.gallery.nft_limit'),
             ),
-            'nftsPerPage' => $nftsPerPage,
+            'nftsPerPage' => (int) config('dashbrd.gallery.pagination.nfts_per_page'),
+            'collectionsPerPage' => (int) config('dashbrd.gallery.pagination.collections_per_page'),
             'nftLimit' => config('dashbrd.gallery.nft_limit'),
+            'nftCount' => $user->nfts->count(),
```
Comment (crnkovic) [inline app/Http/Controllers/MyGalleryController.php:76]: same as above

**Review by crnkovic — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/MyGalleryController.php (hunk 2 above)
Comment (shahin-hq) [reply app/Http/Controllers/MyGalleryController.php]: Ah, $hit, I didn't pay too much attention into this line as I copied it from the method [above](https://github.com/ArdentHQ/dashbrd/pull/370/files#diff-cc0cc28e719e766a0e67369768c8a629c7b2201759058408ed67d22ca0ba1c44R30), fixed all instances.

**Comment by ItsANameToo** [comment]
some conflicts in the meantime @shahin-hq

**Review by alfonsobries — APPROVED** [review]

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**